### PR TITLE
chore: add refactor issue form

### LINKED ISSUE #1900: feat: add issue form for refactor code
### What feature?

I need to refactor the code, but the problem is which issue form to choose. Since it is not a new feature, creating a new issue form will likely be a good choice. 

I will ensure it follows proper conventions and standards and avoid unnecessary fields.

### Add screenshots

> Similar to this
image

### Record

- [X] I have checked the existing [issues](https://github.com/priyankarpal/ProjectsHut/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/priyankarpal/ProjectsHut/blob/main/contributing.md)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/priyankarpal/ProjectsHut/blob/main/CODE_OF_CONDUCT.md)
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Anmol-Baranwal**
## Related Issue

Closes #1900

## Description

- You need to add this label: `goal: refactor`

## Screenshots

image

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/refactor_code.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Refactor Code 🔧
+description: Use this label for code refactoring tasks.
+title: "[Refactor] <write what you want to add>"
+labels: ["goal: refactor", "🚦status: awaiting triage"]
```
Comment (priyankarpal) [inline .github/ISSUE_TEMPLATE/refactor_code.yml:4]: change the label to `🛠 goal: refactor`
```suggestion
labels: ["🛠 goal: refactor", "🚦status: awaiting triage"]
```

**Review by priyankarpal — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/refactor_code.yml (hunk 1 above)
Comment (Anmol-Baranwal) [reply .github/ISSUE_TEMPLATE/refactor_code.yml]: Done.

**Review by priyankarpal — APPROVED** [review]

**Comment by priyankarpal** [comment]
Hey @Anmol-Baranwal  ! 👋

For secure collaboration, remember to check out GitHub's commit signature verification feature. It helps ensure the authenticity of commits and prevents tampering.

Learn how it works: [GitHub Docs](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)

Happy coding! 😄

**Comment by Anmol-Baranwal** [comment]
@priyankarpal 
You didn't add the labels, and also like can you tell how to sign off verified commits. Apart from the settings on GitHub which I already read on docs.

**Comment by priyankarpal** [comment]
> @priyankarpal You didn't add the labels, and also like can you tell how to sign off verified commits. Apart from the settings on GitHub which I already read on docs.

you have generate a GPG key after you need to add that key to github

**Pull request merged**